tests/validation_v2/upgrade/rancher_upgrade_test.py

```diff
@@ -0,0 +1,117 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-b', help='base version')
+    parser.add_argument('-t', help='target version')
+    parser.add_argument('-s', help='server node')
+    parser.add_argument('-u', help='ssh username of rancher server host')
+    args = parser.parse_args()
+
+    tmp_dir = os.path.join(root_dir, 'tmp')
+    logger.info("tmp_dir is: %s", tmp_dir)
+
+    core_dir = os.path.join(root_dir,'tests','validation','cattlevalidationtest','core')
+    logger.info("core_dir is: %s", core_dir)
+
+    core_target_checkedout_dir = os.path.join(tmp_dir,'rancher-tests','tests','validation','cattlevalidationtests','core')
+    logger.info("core_target_checkedout_dir is: %s", core_target_checkedout_dir)
+
+    core_target_dir = os.path.join(root_dir,'tests','validation','cattlevalidationtest','core_target')
+    logger.info("core_target_dir is: %s", core_target_dir)
+
+    upgrade_test(args.b, args.t, args.s, args.u, tmp_dir, core_dir, core_target_dir, core_target_checkedout_dir)
+
+
+def upgrade_test(base, target, servernode, username, tmp_dir, core_dir, core_target_dir, core_target_checkedout_dir):
+    logger.info("CREATING SERVICES NOW IN BASE SETUP...")
+    create_cmd = "py.test "+core_dir+"/ -v -m create -s"
+    logger.info("create command is: %s", create_cmd)
+
+    os.system(create_cmd)
+    time.sleep(5)
+
+    upgrade_rancher_server(base, target, servernode, username)
+    #below one line is temporary until we start having real tagged versions
+    os.system(("cp -r "+core_dir+"/*.py "+tmp_dir))
+    #os.system("git clone -b master --single-branch https://github.com/aruneli/rancher-tests.git")
+    logger.info("COPYING TARGET LIBRARIES in core_target folder...")
+
+    os.system(("cp -r "+tmp_dir+"/*.py "+core_target_dir))
+    logger.info("VALIDATING UPGRADED SETUP NOW WITH TARGET")
+
+    validate_cmd = "py.test "+core_target_dir+"/ -v -m validate -s"
+    logger.info(validate_cmd)
+
+    os.system(validate_cmd)
+    logger.info("VALIDATION COMPLETE")
+
+    os.system("rm -rf "+tmp_dir+"/*")
+    os.system("rm -rf "+core_target_dir+"/*")
+
+
+def upgrade_rancher_server(base, target, servernode, username):
+    logger.info("UPGRADING RANCHER SERVER TO TARGET")
+
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    privatekeyfile = os.path.expanduser('~/.ssh/google_compute_engine')
+    mykey = paramiko.RSAKey.from_private_key_file(privatekeyfile)
+
+    ssh.connect(servernode, username=username, pkey=mykey)
+    stdin, stdout, stderr = ssh.exec_command("sudo docker ps")
+    response = stdout.readlines()
+    logger.info(response)
+
+    stdin, stdout, stderr = ssh.exec_command("sudo docker stop $(sudo docker ps -q | awk '{print $1}')")
+    server_container_id = stdout.readlines()[0].strip("\n")
+    logger.info(server_container_id)
+
+    stdin, stdout, stderr = ssh.exec_command("sudo docker ps -a | awk ' NR>1 {print $2}' | cut -d \: -f 2 | cut -d \\v -f 2")
```

Comment:
This command and the command above it is making some brittle assumptions about what containers are running on the host. Would they break if the rancher-server container was ever not the first container listed in docerk ps?

Also, where are we creating the rancher-server container? Or do you assume that it has been pre-created? I was thinking that if you are creating the container, you can name it something explicit and then find it by name or id in the upgrade requests.